agent/internal/client/mock.go

```diff
@@ -343,6 +343,9 @@ func (c *Mock) GetLoggerProducer(ctx context.Context, tsk *task.Task, _ *LoggerC
 	// in tests, otherwise the append line function will try to access the
 	// task ID in the task pointer every time `sendTaskLogLine` is called.
 	taskID := tsk.Id
+
+	// Every time we make a new producer, we should flush it between tests.
+	c.flushTaskLog(taskID)
```

Comment:
This was a bug in our testing where all our logs were persisting through tests, causing false positives. I only discovered it because I ran patches that just prints all the 'task' logs, but it printed a load from previously ran tests that used the same taskID.

Reply:
nice catch!